src/Time/Units.hs

```diff
@@ -162,11 +176,19 @@ instance KnownUnitName unit => Read (Time unit) where
         readP :: ReadP (Time unit)
         readP = do
             let naturalP = read <$> munch1 isDigit
-            n <- naturalP
-            m <- option 1 (char '/' *> naturalP)
+            -- If a '+' is parsed as part of a mixed fraction, the other parts
+            -- are no longer optional.  This separation is required to prevent
+            -- e.g. "3+2" successfully parsing.
+            let fullMixedExpr = (,,) <$> (naturalP <* char '+')
+                                     <*> (naturalP <* char '/')
+                                     <*> naturalP
+            let improperExpr = (,,) 0 <$> naturalP
```

Comment:
Using `-XTupleSections` extension you can write this as:

```haskell
let improperExpr = (0,,) <$> naturalP
                         <*> option 1 (char '/' *> naturalP)
```

Just a minor suggestions 🙂